src/app/public/modules/i18n/currency/currency-format.service.ts

```diff
@@ -0,0 +1,118 @@
+import { getCurrencySymbol } from '@angular/common';
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { map, take } from 'rxjs/operators';
+
+import { SkyCurrencyFormat } from './currency-format';
+import { SkyAppLocaleProvider } from '../locale-provider';
+
+type IsoCodeAndLocale = { isoCode: string; locale: string };
+type CurrencyFormatParts = {
+  symbol: string;
+  symbolLocation: 'p' | 's';
+  decimalCharacter: string;
+  groupCharacter: string;
+};
+
+/**
+ * Service to get a currency's format given an iso code.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class SkyCurrencyFormatService {
+  constructor(private localeProvider: SkyAppLocaleProvider) { }
+
+  /**
+   * Gets the currency formatting.
+   * @param params optional params object
+   * @param params.isoCode the ISO 4217 Code. Defaults to `USD`.
+   * @param params.locale the locale.
+   */
+  public getCurrencyFormat(params?: Partial<IsoCodeAndLocale>): SkyCurrencyFormat {
```

Comment:
SkyAppLocaleProvider returns an observable (bc users can override it) so I think providing a non-observable-locking call could be valuable.

Defintiely open to opionions though.

Reply:
Removed SkyAppLocaleProvider so this isn't an issue